tests/e2e/common-operator-integ-suite.sh

```diff
@@ -191,18 +214,50 @@ if [ "${SKIP_BUILD}" == "false" ]; then
     get_internal_registry
   fi
 
-  # BUILD AND PUSH IMAGE
+  # BUILD AND PUSH OPERATOR IMAGE
   build_and_push_image
 
-  if [ "${OCP}" == "true" ]; then
-    # This is a workaround
-    # To avoid errors of certificates meanwhile we are pulling the operator image from the internal registry
-    # We need to set image $HUB to a fixed known value after the push
-    # This value always will be equal to the svc url of the internal registry
-    HUB="image-registry.openshift-image-registry.svc:5000/sail-operator"
+  # If OLM is enabled, deploy the operator using OLM
+  if [ "${OLM}" == "true" ]; then
+    # Set the platform to Kubernetes by default.
+    # We are skipping the deploy via OLM test on OCP because the workaround to avoid the certificate issue is not working.
+    HELM_PLATFORM="kubernetes"
+    
+    # Install OLM in the cluster because it's not available by default in kind.
+    ${OPERATOR_SDK} olm install
+
+    # Set image-related variables
+    IMAGE_TAG_BASE="${HUB}/${IMAGE_BASE}"
+    BUNDLE_IMG="${IMAGE_TAG_BASE}-bundle:v${VERSION}"
+
+    # Deploy the operator using OLM
+    IMAGE="${HUB}/${IMAGE_BASE}:${TAG}" \
+    IMAGE_TAG_BASE="${IMAGE_TAG_BASE}" \
+    BUNDLE_IMG="${BUNDLE_IMG}" \
+    HELM_PLATFORM="${HELM_PLATFORM}" \
+    make bundle bundle-build bundle-push
+
+    # Create operator namespace
+    ${COMMAND} create ns "${NAMESPACE}" || echo "namespace ${NAMESPACE} already exists"
```

Comment:
The msg can be misleading when the `oc create ns` actually fails because of something else.

Reply:
Changed